source/_addonStore/dataManager.py

```diff
@@ -89,15 +100,21 @@ def __init__(self):
 		self._compatibleAddonCache = self._getCachedAddonData(self._cacheCompatibleFile)
 		self._installedAddonsCache = _InstalledAddonsCache()
 		# Fetch available add-ons cache early
-		threading.Thread(
+		self._getAddonsThread = threading.Thread(
 			target=self.getLatestCompatibleAddons,
 			name="initialiseAvailableAddons",
-		).start()
+			daemon=True,
+		)
+		self._getAddonsThread.start()
+
+	def terminate(self):
+		if self._getAddonsThread.is_alive():
+			self._getAddonsThread.join(timeout=FETCH_TIMEOUT_S + 1)

```

Comment:
Having NvDA hang on exit for up to 120 seconds does not sound that great.
In deed, restarting NVDA, the new NvDA would kill it off after 10 seconds anyway.
What would be the consequence of not joining the thread here, and just having it killed off (as it is now a daemon thread) on process exit?
If it is important that we have the ability to clean up after requests/get etc returns on terminate, then we could look at putting every requests.get call in its own daemon thread which we would normally block on, but would have the ability to abandon if necessary.

Reply:
I've changed this timeout to 1. It's probably fine not to join, but if we reset config and reinitialize submodules the thread should be killed before firing the next one on initialization.